esp/services/ldapenvironment/ldapenvironmentService.cpp

```diff
@@ -365,25 +374,45 @@ bool CldapenvironmentEx::onLDAPCreateEnvironment(IEspContext &context, IEspLDAPC
         if (req.getCreateK8sSecrets())
             createSecret(ST_K8S, respLDAPAdminSecretName.str(), respLDAPAdminUser.str(), respLDAPAdminPwd.str(), notes);
         if (req.getCreateVaultSecrets())
-            createSecret(ST_VAULT, respLDAPAdminSecretName.str(), respLDAPAdminUser.str(), respLDAPAdminPwd.str(), notes);
+            createSecret(ST_AUTHN_VAULT, respLDAPAdminSecretName.str(), respLDAPAdminUser.str(), respLDAPAdminPwd.str(), notes);
 
         //----------------------------------
         // Set response
         //----------------------------------
-        VStringBuffer ldapcredskey("ldapcredskey-%s", req.getEnvName());
-        VStringBuffer hpcccredskey("hpcccredskey-%s", req.getEnvName());
+        StringBuffer ldapcredskey;
+        StringBuffer hpcccredskey;
+        if (respVaultID.isEmpty())
+        {
+            //No vault, create K8s authn key name
+            ldapcredskey.appendf("ldapcredskey-%s", req.getEnvName());
+            hpcccredskey.appendf("hpcccredskey-%s", req.getEnvName());
+        }
+        else
+        {
+            //Vault, specify the secret name as the key
+            ldapcredskey.set(respLDAPAdminSecretName.str());
+            hpcccredskey.set(respHPCCAdminSecretName.str());
+        }
         ldapcredskey.toLowerCase();
         hpcccredskey.toLowerCase();
 
         resp.setLDAPAdminUsername(respLDAPAdminUser.str());
         resp.setLDAPAdminPassword(respLDAPAdminPwd.str());
         resp.setHPCCAdminUsername(respHPCCAdminUser.str());
         resp.setHPCCAdminPassword(respHPCCAdminPwd.str());
-        VStringBuffer ldapHelm("\n\n"
-                               "secrets:\n"
+
+        StringBuffer helmSecrets;
+        if (req.getCreateK8sSecrets())
+        {
+            helmSecrets.appendf("secrets:\n"
```

Comment:
Should you have a similar section for vaults that shows how the vault is set up in the values.yaml?

i.e. what the following should look like?:

```
vaults:
  authn:
    - name: my-authn-vault
      url: http://${env.VAULT_SERVICE_HOST}:${env.VAULT_SERVICE_PORT}/v1/secret/data/authn/${secret}
      kind: kv-v2

```